.github/workflows/ci.yml

```diff
@@ -42,35 +42,13 @@ jobs:
         name: Checkout
         uses: actions/checkout@v3
       -
-        name: Install pipenv
-        run: |
-          pipx install pipenv==2022.10.12
-      -
-        name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.9
-          cache: "pipenv"
-          cache-dependency-path: 'Pipfile.lock'
-      -
-        name: Install dependencies
-        run: |
-          pipenv sync --dev
-      -
-        name: List installed Python dependencies
-        run: |
-          pipenv run pip list
-      -
-        name: Make documentation
-        run: |
-          cd docs/
-          pipenv run make html
-      -
-        name: Upload artifact
-        uses: actions/upload-artifact@v3
-        with:
-          name: documentation
-          path: docs/_build/html/
+        name: Deploy docs
```

Comment:
So integrating it here means this runs for every push on every branch + pull requests.  Does this action version the page?  Or maybe it's a pages setting to only pull from 1 branch?

Reply:
Yea I don't know the best way to do this, def would need your help. I noticed that. There are other ways to do it where eg you just build the docs with python sinilar to how it works now but I don't k ow about the pushing to gh-pages part.

I also thought about have a test which builds the docs (eg with python) on push/PR but use this one only on commits to main.

Again, would be grateful for your expertise (whenever you have the time)

Reply:
I looked into this some.  Currently, a push or PR would update the branch `gh-pages`, which we'd select as the branch for pages to deploy from.  So the actual deploy should be `main` and we can restirct it to only related changes.

For PRs/pushes, `mkdocs build --config-file "${CONFIG_FILE}" --force` should suffice to check the docs build.  And there's also grabbing the built files into an artifact.

Not sure when I'll be able to help on this, but that's the gist of the work I think is needed

Reply:
Ok no worries, I'll see if I can figure it out in the meantime, take your time. Appreciate the roadmap!

Reply:
I made a stab at this, let me know any thoughts—again, if / when you have some time. Two jobs `documentation` (replaces the old one just now with `mkdocs build`) and `documentation-deploy` which runs `if: github.event_name == 'push' && github.ref == 'refs/heads/main'`

Reply:
That looks fine to me. Thanks for doing it